### DISCUSSION TITLE
[FEATURE REQ] Refilter already processed feed items when download slots are available

### DISCUSSION THREAD

**Discussion by ftruzzi** [autobrr/autobrr#1681] [Ideas]
**Is your feature request related to a problem? Please describe.**

Hi, and thank you for your work!

I'm trying to figure out how to use autobrr for the following use case: I have a long RSS feed of freeleech torrents I want to snatch, but I can only download N torrents per day on this tracker. So I set up a filter with "Max N torrents" per day. This works fine and autobrr only downloads N torrents the first day.

The second day, when reparsing the feed, no new torrents are downloaded and I see in the logs:

> DBG found (0) new items to process feed="Last 100 freeleech" module=feed

so I assume all previously fetched items are not going through the filter because they've already been seen, but today the filter should let N more go through?

**Describe the solution you'd like**

Have a feature to not mark an item as "old" if it was rejected from a filter due to a "max downloads rule"

**Describe alternatives you've considered**

n/a

**Additional context**

n/a

**Comment by zze0s** [comment]
That's a pretty specific feature req :thinking: 

Does your feed remove the downloaded torrents? In that case you'll have to manually clear the feed-cache each day and trigger a run.

Maybe we should add an option to skip adding items to the cache. But your feed would still refresh at your given interval.

The filter checks are only triggered if the feed check finds new items.

**Comment by KyleSanderson** [comment]
> That's a pretty specific feature req 🤔
> 
> Does your feed remove the downloaded torrents? In that case you'll have to manually clear the feed-cache each day and trigger a run.
> 
> Maybe we should add an option to skip adding items to the cache. But your feed would still refresh at your given interval.
> 
> The filter checks are only triggered if the feed check finds new items.

They're looking for *arr queueing functionality - which autobrr isn't built for today. It's a sneaky request, but there's a ton of work to support this "properly" (maintaining/showing the queue). The cheap ask is what they described, which is also doable, but you'd still need a way to ignore already approved items, which goes back to unique.

**Comment by ftruzzi** [comment]
Thanks for your replies!

> Does your feed remove the downloaded torrents? In that case you'll have to manually clear the feed-cache each day and trigger a run.

Unfortunately it doesn't remove the entries, so after clearing the cache it tries to re-download the already existing N torrents and stops after that.

> Maybe we should add an option to skip adding items to the cache.

> there's a ton of work to support this "properly" (maintaining/showing the queue) [...] you'd still need a way to ignore already approved items

What if there was a flag on the feed to "only cache downloaded entries"? Would it make sense / be easy enough to implement?
